### PR TITLE
Per world ticks per spawn settings

### LINKED ISSUE #6838: Per-world/paper.yml ticks-per.entity-spawns
### Is your feature request related to a problem?

I want my nether world's ticks-per.monster-spawns faster to replicate vanilla zombie piglin farm behavior, as they are much less efficient in my server; without negatively affecting the overworld.

### Describe the solution you'd like.

Like the spawn-limits category of bukkit.yml being in paper, just with ticks-per.spawns.
```
ticks-per:
  water-underground-creature-spawns: -1
  animal-spawns: -1
  monster-spawns: -1
  water-spawns: -1
  water-ambient-spawns: -1
  ambient-spawns: -1
  ```

### Describe alternatives you've considered.

Raising ticks-per.monster-spawns in bukkit.yml. This has the effect of causing lag for the overworld as opposed to optimized values, instead of just helping with the nether spawns.

### Other

_No response_

### PR DESCRIPTION
**Pull request by Machine-Maker**
Fixes #6838

### CODE REVIEW

File: patches/server/0844-Per-world-ticks-per-spawn-settings.patch [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index 0906a2d410e11cb352b13e5c0390560085b8f254..c88244aa5212da4789b711a978e4ee823ca887cb 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -764,6 +764,14 @@ public class PaperWorldConfig {
+         }
+     }
+ 
++    public Reference2IntMap<MobCategory> perWorldTicksPerSpawns = new Reference2IntOpenHashMap<>(NaturalSpawner.SPAWNING_CATEGORIES.length);
```
Comment (kennytv) [inline patches/server/0844-Per-world-ticks-per-spawn-settings.patch]: ~~Use an enummap~~ boxing is a thing, right

File: patches/server/0844-Per-world-ticks-per-spawn-settings.patch [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
++    private void setPerWorldTicksPerSpawns() {
++        perWorldSpawnLimits.defaultReturnValue(-1);
++        for (final MobCategory category : NaturalSpawner.SPAWNING_CATEGORIES) {
++            perWorldSpawnLimits.put(category, getInt("ticks-per-spawn." + category.getName(), perWorldTicksPerSpawns.getInt(category)));
```
Comment (kennytv) [inline patches/server/0844-Per-world-ticks-per-spawn-settings.patch:19]: lowercase category name

Comment (kennytv) [inline patches/server/0844-Per-world-ticks-per-spawn-settings.patch]: Wrong map used

**Review by kennytv — CHANGES_REQUESTED** [review]

**Review by electronicboy — APPROVED** [review]

**Comment by Machine-Maker** [comment]
Rebased for 1.19

**Review by kennytv — APPROVED** [review]

**Review by Owen1212055 — APPROVED** [review]

**Pull request merged**